### DISCUSSION TITLE
Export only selected columns

### DISCUSSION THREAD

**Discussion by RuLeiFei** [rappasoft/laravel-livewire-tables#1226] [Q&A] [answered]
Hello, I am studying the package with great interest and pleasure, but I have some questions that I still cannot get an answer from the documentation. In the example of the export function, all columns from getSelected get into the file, but I did not find a method that would transfer only the selected columns. There is an array in selectedColumns, but it is passing slugs. Is there any method to pass only selected columns?

Thanks for the future answer!

**Comment by lrljoe** [comment]
I'm assuming you've built a custom export class, in which case you can pass that array into the constructor.

**Comment by RuLeiFei** [comment]
Yes, you are absolutely correct, but I apparently formulated my question incorrectly. I could not get the names of the selected columns as in the database to pass them on, but only slugs. But it's worth reading the documentation for more details. In any case, thanks!

**Reply by lrljoe** [reply]
If you're still stuck after reading the docs then let me know, as I'm a tad confused as to what you're trying to achieve/use case, which is probably just me skim reading to be honest!

Passing the selected columns should give you what you're looking for,  but may not if you're using custom slugs for the column that are set to something odd!

**Reply by RuLeiFei** [reply]
Thank you very much.

For example, I want to pass an array of selected columns to the export function, which would then use `->select()`  to limit the data to be exported. If I do this `$columns = $this->selectedColumns`.
I get the name of the selected columns of the form  `$key => $slug`, so I need the selected columns like in the database. I looked at how the package is arranged, and the array formed through the `getSlug() ` method, if you have a solution how to get the names of the selected columns as they are specified in the database, I will be grateful .

**Reply by lrljoe** [reply, marked as answer]
To reverse a slug, you'll need the Str helper included, but this should work:

```php

Str::title(str_replace('-', ' ', $slug)

```

If not, I'll take a closer look later this week.

Otherwise, in the interim, just have an array mapping the slug to proper column name in your export class, and pluck out the proper column names for your select.

**Reply by RuLeiFei** [reply]
Thanks for your answer, I did as you advised: in the function I got a list of all columns and compared with the list of selected ones using Laravel methods.
